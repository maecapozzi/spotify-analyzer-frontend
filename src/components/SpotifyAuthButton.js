import React, { Component } from 'react'

class SpotifyAuthButton extends Component {
  constructor () {
    super()

    let uri = process.env.API_URI || 'http://0.0.0.0:3001'
    this.url = `${uri}/auth/spotify`
  }

  render () {
    return (
      <div className='login-container'>
        <button className='login-button'>
          <a className='login-button' href={this.url}>Login with Spotify</a>
        </button>
      </div>
    )
  }
}

export default SpotifyAuthButton
