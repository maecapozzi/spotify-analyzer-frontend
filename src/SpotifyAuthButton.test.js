import React from 'react'
import SpotifyAuthButton from './components/SpotifyAuthButton'
import renderer from 'react-test-renderer'

test('it renders correctly', () => {
  const component = renderer.create(
    <SpotifyAuthButton />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
