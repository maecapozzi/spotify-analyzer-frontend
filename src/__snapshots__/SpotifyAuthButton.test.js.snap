// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<button>
  <a
    href="http://localhost:3001/auth/spotify"
  >
    Login with Spotify
  </a>
</button>
`;
